### PR TITLE
Port to windows and Add windows build test to modmesh_viewer CI workflow

### PR DESCRIPTION
**Pull request by tychuang1211**
#106 

1. Support building with MSVC
2. Add OS `windows-2022` to CI workflow

### CODE REVIEW

File: .github/workflows/modmesh_viewer.yml [hunk 1]
```diff
@@ -51,10 +56,27 @@ jobs:
         pip3 install -U setuptools
         pip3 install -U numpy pytest flake8
 
+    - name: Install Qt (windows)
+      if: runner.os == 'Windows'
```
Comment (yungyuc) [inline .github/workflows/modmesh_viewer.yml]: Is there a reason not to use ``${{ startsWith(matrix.os, 'windows') }}``?

File: cpp/modmesh/base.hpp [hunk 2]
```diff
@@ -47,6 +47,10 @@
 
 namespace modmesh
 {
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
```
Comment (yungyuc) [inline cpp/modmesh/base.hpp]: Is it a right way to include in a namespace?

File: cpp/modmesh/spacetime/core.hpp [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 #include <memory>
 #include <vector>
+#include <functional>
```
Comment (yungyuc) [inline cpp/modmesh/spacetime/core.hpp:10]: Thanks for catching this!

File: .github/workflows/modmesh_viewer.yml [hunk 4]
```diff
@@ -15,8 +15,9 @@ jobs:
 
     strategy:
         matrix:
-          os: [ubuntu-22.04, macos-12]
+          os: [ubuntu-22.04, macos-12, windows-2022]
           cmake_build_type: [Release, Debug]
+          arch: [amd64]
```
Comment (yungyuc) [inline .github/workflows/modmesh_viewer.yml]: Do we need it?  Isn't the cloud runner always uses x86 (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)?

**Review by yungyuc — CHANGES_REQUESTED** [review]
This is exciting!  wakuwaku

File: .github/workflows/modmesh_viewer.yml (hunk 1 above)
Comment (tychuang1211) [reply .github/workflows/modmesh_viewer.yml]: Actually, no. Well, except for I think it's easier to read.

Comment (yungyuc) [reply .github/workflows/modmesh_viewer.yml]: In that case, could you use the original way to test it?  `startsWith` reads more accurate semantically.

File: .github/workflows/modmesh_viewer.yml (hunk 4 above)
Comment (tychuang1211) [reply .github/workflows/modmesh_viewer.yml]: Hmm, I don't really know about this. It's the input to msvc-dev-cmd and the default value seems sufficient, I think we could just delete the line.

Comment (yungyuc) [reply .github/workflows/modmesh_viewer.yml]: Agree.  If it's not needed, let's delete it.

File: cpp/modmesh/base.hpp (hunk 2 above)
Comment (tychuang1211) [reply cpp/modmesh/base.hpp]: Come to think of it, we should put this outside the namespace. XD

File: .github/workflows/modmesh_viewer.yml [hunk 5]
```diff
@@ -26,6 +27,10 @@ jobs:
       with:
         fetch-depth: 1
 
+    - uses: ilammy/msvc-dev-cmd@v1
+      with:
+        arch: ${{ matrix.arch }}
```
Comment (tychuang1211) [inline .github/workflows/modmesh_viewer.yml]: We'll also need to delete these two lines.

File: .github/workflows/modmesh_viewer.yml (hunk 1 above)
Comment (tychuang1211) [reply .github/workflows/modmesh_viewer.yml]: No problem.

File: cpp/modmesh/base.hpp (hunk 2 above)
Comment (yungyuc) [reply cpp/modmesh/base.hpp]: The whole fenced code section can be moved out.

**Review by yungyuc — APPROVED** [review]

**Comment by yungyuc** [comment]
This is wonderful.  Thank you so much, @tychuang1211

**Pull request merged**